src/test_l2_dsCompositeIn.c

```diff
@@ -1,119 +1,307 @@
- *
- */
+* @file test_l2_dsCompositeIn.c
+* @page dsCompositeIn Level 2 Tests
+*
+* ## Module's Role
+* This module includes Level 2 functional tests (success and failure scenarios).
+* This is to ensure that the dsCompositeIn APIs meet the requirements across all vendors.
+*
+* **Pre-Conditions:**  None@n
+* **Dependencies:** None@n
+*
+* Ref to API Definition specification documentation : [halSpec.md](../../../docs/halSpec.md)
+*/
 
-/**
- * @addtogroup Device_Settings Device Settings Module
- * @{
- */
+#include <ut.h>
+#include <ut_log.h>
+#include <stdlib.h>
+#include <time.h>
+#include "dsCompositeIn.h"
 
 /**
- * @addtogroup Device_Settings_HALTEST Device Settings HAL Tests
- * @{
- */
+* @brief Test to verify the number of composite input ports
+*
+* This test verifies the number of composite input ports by comparing the expected number of ports with the actual number of ports returned by the API. The test is designed to ensure that the API is correctly identifying the number of ports and that the system is correctly configured.
+*
+* **Test Group ID:** 02@n
+* **Test Case ID:** 001@n
+*
+* **Test Procedure:**
+* Refer to UT specification documentation [l2_dsCompositeIn_test_specification.md](l2_dsCompositeIn_test_specification.md)
+*/
 
-/**
- * @addtogroup DS_CompositeIn_HALTEST Device Settings Composite Input HAL Tests
- * @{
- */
+void test_l2_dsCompositeIn_VerifyCompositeInputPorts(void)
+{
+    UT_LOG("Entering test_l2_dsCompositeIn_VerifyCompositeInputPorts...");
+
+    dsError_t status;
+    uint8_t numInputs;
+    uint8_t expectedNumInputs = 2; // This value should be fetched from the configuration file
+
+    // Step 1: Call the pre-requisite API dsCompositeInInit to initialize the COMPOSITE Input module.
+    status = dsCompositeInInit();
+    UT_ASSERT_EQUAL(status, dsERR_NONE);
+    if (status != dsERR_NONE)
+    {
+        UT_LOG("dsCompositeInInit failed with status: %d", status);
+        return;
+    }
+
+    // Step 2: Call the API dsCompositeInGetNumberOfInputs to get the number of COMPOSITE Input ports on the specific platform.
+    status = dsCompositeInGetNumberOfInputs(&numInputs);
+    UT_LOG("Invoked dsCompositeInGetNumberOfInputs() . Return status: %d, numInputs : %d ", status, numInputs);
+    // Step 3: Check the return status of the API. If it is not dsERR_NONE, then the API has failed. The test case should fail in this scenario.
+    UT_ASSERT_EQUAL(status, dsERR_NONE);
+    if (status != dsERR_NONE)
+    {
+        UT_LOG("dsCompositeInGetNumberOfInputs failed with status: %d", status);
+        // Call the post-requisite API dsCompositeInTerm to terminate the COMPOSITE Input module.
+        dsCompositeInTerm();
+        return;
+    }
+
+    // Step 4: If the return status is dsERR_NONE, compare the value of the uint8_t variable with the expected number of COMPOSITE input ports from the configuration file.
+    UT_ASSERT_EQUAL(numInputs, expectedNumInputs);
+    if (numInputs != expectedNumInputs)
```

Comment:
this if condition is not required as the previous line assert will check the condition

Reply:
this is to ensure that test exits after cleanup on assertion failure, as agreed on previous discussions